sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/optimizer/FilterPushdownSuite.scala

```diff
@@ -1384,4 +1384,14 @@ class FilterPushdownSuite extends PlanTest {
       condition = Some("x.a".attr === "z.a".attr)).analyze
     comparePlans(optimized, correctAnswer)
   }
+
+  test("SPARK-28220: Push down the foldable predicate to both sides of Join") {
+    val x = testRelation.subquery('x)
+    val y = testRelation.subquery('y)
+    val originalQuery = x.join(y, condition = Some(false))
```

Comment:
How about we make sure the join condition contains no constant? If it's true, we can remove it. If it's false, we can remove the join and return an empty relation.

Reply:
If it is false, if it is an inner join, it can be removed if it is an inner join, but it cannot be removed if it is a left/right join.

Reply:
for left semi join, we can remove the join, right? for left-anti, we can remove the join condition if it's false.

Reply:
OK. Add a new rule? The constant condition may occur at any time. Example from SPARK-28220:
```sql
select * from table1 left join (select d, 'w1' as r from table2) on a = d and r = 'w2' where b = 2
```

Reply:
yea adding a new rule to optimize constants in join conditions, then we don't need to pushdown it.

Reply:
ok